src/scrapers/leumi.js

```diff
@@ -0,0 +1,176 @@
+import moment from 'moment';
+import { BaseScraperWithBrowser, LOGIN_RESULT } from './base-scraper-with-browser';
+import { dropdownSelect, fillInput, clickButton, waitUntilElementFound } from '../helpers/elements-interactions';
+import { waitForNavigation } from '../helpers/navigation';
+import { SHEKEL_CURRENCY, NORMAL_TXN_TYPE } from '../constants';
+
+const BASE_URL = 'https://hb2.bankleumi.co.il/';
+const DATE_FORMAT = 'DD/MM/YY';
+
+function getTransactionsUrl() {
+  return `${BASE_URL}/ebanking/Accounts/ExtendedActivity.aspx?WidgetPar=1#/`;
+}
+
+function getPossibleLoginResults() {
+  const urls = {};
+  urls[LOGIN_RESULT.SUCCESS] = /ebanking\/SO\/SPA.aspx/;
+  urls[LOGIN_RESULT.INVALID_PASSWORD] = /InternalSite\/CustomUpdate\/leumi\/LoginPage.ASP/;
+  // urls[LOGIN_RESULT.CHANGE_PASSWORD] = ``; // TODO should wait until my password expires
+  return urls;
+}
+
+function createLoginFields(credentials) {
+  return [
+    { selector: '#wtr_uid', value: credentials.username },
+    { selector: '#wtr_password', value: credentials.password },
+  ];
+}
+
+function getAmountData(amountStr) {
+  const amountStrCopy = amountStr.replace(',', '');
+  const amount = parseFloat(amountStrCopy);
+  const currency = SHEKEL_CURRENCY;
+
+  return {
+    amount,
+    currency,
+  };
+}
+
+function convertTransactions(txns) {
+  return txns.map((txn) => {
+    const txnDate = moment(txn.date, DATE_FORMAT).toISOString();
+
+    const credit = getAmountData(txn.credit).amount;
+    const debit = getAmountData(txn.debit).amount;
+    const amount = (Number.isNaN(credit) ? 0 : credit) - (Number.isNaN(debit) ? 0 : debit);
+    return {
+      type: NORMAL_TXN_TYPE,
+      identifier: txn.reference ? parseInt(txn.reference, 10) : null,
+      date: txnDate,
+      processedDate: txnDate,
+      originalAmount: amount,
+      originalCurrency: SHEKEL_CURRENCY,
+      chargedAmount: amount,
+      description: txn.description,
+      /* memo: txn.memo, TODO add this line to export transaction memo */
+    };
+  });
+}
+
+async function fetchTransactionsForAccount(page, startDate) {
+  // TODO provide actual start date
+  await dropdownSelect(page, 'select#ddlTransactionPeriod', '004');
+  await waitUntilElementFound(page, 'select#ddlTransactionPeriod');
+  await fillInput(
+    page,
+    'input#dtFromDate_textBox',
+    startDate.format('DD/MM/YY'),
+  );
+  await clickButton(page, 'input#btnDisplayDates');
+  await waitForNavigation(page);
+  await waitUntilElementFound(page, 'table#WorkSpaceBox table#ctlActivityTable');
+  await clickButton(page, 'a#lnkCtlExpandAll');
+
+  const selectedSnifAccount = await page.$eval('#ddlAccounts_m_ddl option[selected="selected"]', (option) => {
+    return $(option).text(); // eslint-disable-line no-undef
```

Comment:
@esakal same comment as above, can we get rid of eslint line disable and make it work?

Reply:
I'm using the jQuery library which is available since the bank uses it in this page so I can use it as well. Since it is a global variable I'm getting this warning. 

The only workaround i'm familiar with is to specify it as global for eslint as suggested [here](https://eslint.org/docs/rules/no-undef) but I think in this use case disabling this line is more appropriate because we cannot assume that all the pages will have jQuery in place. 

wdyt?

Reply:
@esakal if we really wanted to allow jQuery in evaluations, we could add it under the globals section of the `eslintrc` file.
But I don't think that's needed, we can probably have something like:
```node
const selectedSnifAccount = await page.$eval('#ddlAccounts_m_ddl option[selected="selected"]', (option) => {
    return option.innerText;
});
```